src/test/java/com/pusher/rest/CryptoTest.java

```diff
@@ -0,0 +1,88 @@
+package com.pusher.rest;
+
+import com.google.gson.FieldNamingPolicy;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.pusher.rest.util.Crypto;
+import com.pusher.rest.util.EncryptedPayload;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Base64;
+
+import static org.junit.Assume.assumeTrue;
+
+public class CryptoTest {
+    private static final Gson BODY_SERIALISER = new GsonBuilder()
+            .setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES)
+            .create();
+
+    private Crypto pc = null;
+    @Before
+    public void initialize() {
+        if(Crypto.cryptoAvailable()) {
+             pc = new Crypto("aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa", BODY_SERIALISER);
+        }
+    }
+    @Test
+    public void testGenerateSharedSecret() {
+        assumeTrue(Crypto.cryptoAvailable());
+        // Check that the secret generation is generating consistent secrets
+        byte[] sharedSecret = pc.generateSharedSecret("private-encrypted-channel-a");
+        String sharedSecretB64 = Base64.getEncoder().encodeToString(sharedSecret);
+        String expected = "Rp+wpkNpL89qhqco1JkIG31AVXyU8PUVJBr1B2MvdoA=";
+        Assert.assertEquals(expected, sharedSecretB64);
+
+        // Check that the secret generation is using the channel as a part of the generation
+        byte[] sharedSecret2 = pc.generateSharedSecret("private-encrypted-channel-b");
+        if(sharedSecret == sharedSecret2) {
+            Assert.fail();
+        }
+        Crypto pc2 = new Crypto("bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb", BODY_SERIALISER);
```

Comment:
In the same way you have the comment

```
// Check that the secret generation is using the channel as a part of the generation
```

I think you could mention what you are testing for here, e.g.

```
// Check that the secret generation is using the encryption master key as part of the generation
```